pandas/core/tools/datetimes.py

```diff
@@ -913,7 +917,10 @@ def to_datetime(
     else:
         result = convert_listlike(np.array([arg]), format)[0]
 
-    return result
+    #  error: Incompatible return value type (got "Union[Timestamp, NaTType,
+    # Series, Index]", expected "Union[DatetimeIndex, Series, float, str,
+    # NaTType, None]")
+    return result  # type: ignore[return-value]
```

Comment:
maybe use returns instead of assignment to result or change the type declaration of result to match the return type.